tests/bgp/test_bgp_session_flap.py

```diff
@@ -0,0 +1,204 @@
+'''This script is to test BGP session flapping on SONiC and monitor
+the CPU.
+'''
+import logging
+
+import pytest
+import time
+from tests.common.utilities import InterruptableThread
+import textfsm
+import traceback
+
+from natsort import natsorted
+
+logger = logging.getLogger(__name__)
+max_wait = 0
+wait_time = 2
+stop_threads = False
+proc_textfsm = "./bgp/templates/show_proc_cpu.template"
+bgp_sum_textfsm = "./bgp/templates/bgp_summary.template"
+flap_threads = []
+skip_hosts = []
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+
+def get_cpu_stats(dut):
+    proc_cpu = dut.shell("show processes cpu | head -n 20", module_ignore_errors=True)['stdout']
+    proc_mem = dut.shell("show processes memory | head -n 20", module_ignore_errors=True)['stdout']
+    proc_sum = dut.shell("show processes summary | grep -v '0.0 0.0'", module_ignore_errors=True)['stdout']
+    bgp_cpu = dut.shell("show processes cpu | grep bgp", module_ignore_errors=True)['stdout']
+    bgp_v4_sum = dut.shell("show ip bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    bgp_v6_sum = dut.shell("show ipv6 bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    logger.info("CPU:\n{}\nMemory:\n{}\nSummary:\n{}\nBGP Memory:\n{}\nBGP IPv4:\n{}\nIPv6:\n{}\n"
+                .format(proc_cpu, proc_mem, proc_sum, bgp_cpu, bgp_v4_sum, bgp_v6_sum))
+    with open(proc_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_cpu = fsm.ParseText(proc_cpu)[0]
+
+    with open(bgp_sum_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_ipv4 = fsm.ParseText(bgp_v4_sum)[0]
+        parsed_ipv6 = fsm.ParseText(bgp_v6_sum)[0]
+    data = [float(parsed_cpu[0]), float(parsed_cpu[1]), float(parsed_cpu[2]),
+            float(parsed_ipv4[0]), float(parsed_ipv4[1]), float(parsed_ipv4[2]),
+            float(parsed_ipv6[0]), float(parsed_ipv6[1]), float(parsed_ipv6[2])]
+    return data
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, enum_asic_index):
```

Comment:
`enum_asic_index` will run this test on all asic on the host. can you change to `enum_rand_one_frontend_asic_index`

Reply:
I made this change.